agentlocal/agent_local_test.go

```diff
@@ -108,3 +116,91 @@ func TestServerStatus(t *testing.T) {
 		assert.Equal(t, expected, actual)
 	})
 }
+
+func TestGetZipFile(t *testing.T) {
+	setup := func(t *testing.T) ([]*agentlocalpb.AgentInfo, *mockSupervisor, *mockClient, *config.Config) {
+		agentInfo := []*agentlocalpb.AgentInfo{{
+			AgentId:   "/agent_id/00000000-0000-4000-8000-000000000002",
+			AgentType: inventorypb.AgentType_NODE_EXPORTER,
+			Status:    inventorypb.AgentStatus_RUNNING,
+		}}
+		var supervisor mockSupervisor
+		supervisor.Test(t)
+		supervisor.On("AgentsList").Return(agentInfo)
+		agentLogs := make(map[string][]string)
+		agentLogs[inventorypb.AgentType_NODE_EXPORTER.String()] = []string{
+			"logs1",
+			"logs2",
+		}
+		supervisor.On("AgentsLogs").Return(agentLogs)
+		var client mockClient
+		client.Test(t)
+		client.On("GetServerConnectMetadata").Return(&agentpb.ServerConnectMetadata{
+			AgentRunsOnNodeID: "/node_id/00000000-0000-4000-8000-000000000003",
+			ServerVersion:     "2.0.0-dev",
+		})
+		cfg := &config.Config{
+			ID: "/agent_id/00000000-0000-4000-8000-000000000001",
+			Server: config.Server{
+				Address:  "127.0.0.1:8443",
+				Username: "username",
+				Password: "password",
+			},
+		}
+		return agentInfo, &supervisor, &client, cfg
+	}
+
+	t.Run("test zip file", func(t *testing.T) {
+		_, supervisor, client, cfg := setup(t)
+		defer supervisor.AssertExpectations(t)
+		defer client.AssertExpectations(t)
+		ringLog := storelogs.New(10)
+		s := NewServer(cfg, supervisor, client, "/some/dir/pmm-agent.yaml", ringLog)
+		_, err := s.Status(context.Background(), &agentlocalpb.StatusRequest{GetNetworkInfo: false})
+		require.NoError(t, err)
+		// fmt.Sprintf("%v %v", actual, err)
+		rec := httptest.NewRecorder()
+		req := httptest.NewRequest("GET", "/logs.zip", nil)
+		s.Zip(rec, req)
+		b, err := ioutil.ReadAll(rec.Body)
+		require.NoError(t, err)
+
+		expectedFile, err := generateTestZip(s)
```

Comment:
You are generating zip file that you will later compare with, but what if it fails? The output will be not readable.
I think it would be better to unzip received file, and verify content.
Here is example of equality check failing:
![image](https://user-images.githubusercontent.com/91831719/169696640-0da54fd3-0516-4617-a1fc-6681e1498da7.png)

The failure message is impossible to understand, because it is a diff between two binary files